src/App.tsx

```diff
@@ -16,35 +26,121 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([...goodsFromServer]);
+  const [sortState, setSortState] = useState<{
+    reversed: boolean;
+    criterion: SortType;
+  }>({
+    reversed: false,
+    criterion: SortType.None,
+  });
+  const [isReset, setIsReset] = useState(false as boolean);
```

Comment:
```suggestion
  const hasResetButton = sortState.iisReversed || Boolean(sortState.sortType);
```
- It's a dynamic value, you don't need to store it in a state
- Avoid `as`